ml-agents/mlagents/trainers/meta_curriculum.py

```diff
@@ -38,22 +41,18 @@ def __init__(self, curriculum_folder, default_reset_parameters):
                     curriculum_folder, curriculum_filename
                 )
                 curriculum = Curriculum(curriculum_filepath, default_reset_parameters)
+                config_keys: Set[str] = set(curriculum.get_config().keys())
 
                 # Check if any two curriculums use the same reset params.
-                if any(
-                    [
-                        (parameter in curriculum.get_config().keys())
-                        for parameter in used_reset_parameters
-                    ]
-                ):
+                if config_keys & used_reset_parameters:
```

Comment:
This seems cleaner as a set intersection check. Also added some type hints while I was in the neighborhood.